R/FilterStateLogical.R

```diff
@@ -163,11 +168,12 @@ LogicalFilterState <- R6::R6Class( # nolint
       do.call(super$initialize, args)
 
       private$set_choices(choices)
-      if (is.null(selected)) {
-        private$set_selected(TRUE)
-      } else {
-        private$set_selected(selected)
+      if (is.null(selected) && multiple) {
+        selected <- private$choices
+      } else if (is.null(selected)) {
+        selected <- TRUE
```

Comment:
This one gives an error
```r
LogicalFilterState$new(x = c(TRUE, FALSE), selected = c(TRUE, FALSE), multiple = FALSE)
```
```suggestion
      } else if (length(selected) != 1) {
        selected <- TRUE
```

Please include similar test to the one you've just added in test-ChoicesFilterState